### PR TITLE
70 support for rsm plots

### PR DESCRIPTION
**Pull request by ka-sarthak**

### CODE REVIEW

File: src/nomad_measurements/xrd/schema.py [hunk 1]
```diff
@@ -268,31 +299,34 @@ def derive_n_values(self):
...
     )
+    n_values_arr = Quantity(
+        type=np.dtype(np.int64),
+        shape=['*'],
+        derived=lambda self: np.arange(self.n_values),
+        description='Array of integers from 0 to n_values-1',
+    )
     two_theta = Quantity(
-        type=np.dtype(np.float64), shape=['n_values'],
+        type=np.dtype(np.float64), shape=['*'],
```
Comment (hampusnasstrom) [inline src/nomad_measurements/xrd/schema.py:317]: If you do this, there is no need for `derive_n_values()`, `n_values`, or `n_values_arr` (can be removed)

File: src/nomad_measurements/xrd/schema.py [hunk 2]
```diff
@@ -268,31 +299,34 @@ def derive_n_values(self):
...
         a_plot={
-            'x': 'two_theta', 'y': 'intensity'
-        },
-    )
-    q_vector = Quantity(
-        type=np.dtype(np.float64), shape=['*'],
-        unit='meter**(-1)',
-        description='The scattering vector *Q* of the diffractogram',
-        a_plot={
-            'x': 'q_vector', 'y': 'intensity'
```
Comment (hampusnasstrom) [inline src/nomad_measurements/xrd/schema.py:288]: Keep here and rename to `q_norm`

File: src/nomad_measurements/xrd/schema.py [hunk 3]
```diff
@@ -379,9 +627,10 @@ class XRayDiffraction(Measurement):
         | **X-ray Reflectivity (XRR)**                               | Used to study thin film layers, interfaces, and multilayers. Provides info on film thickness, density, and roughness.                                                                                       |
         | **Grazing Incidence X-ray Diffraction (GIXRD)**            | Primarily used for the analysis of thin films with the incident beam at a fixed shallow angle.                                                                                                              |
         ''',
+        # | **Reciprocal Space Mapping (RSM)**                         | High-resolution XRD method to measure a reciprocal space map. Provides additional information used to aid the interpretation of peak displacement, peak broadening or peak overlap.                                                                                                                                                             |
     )
-    results = Measurement.results.m_copy()
-    results.section_def = XRDResult
+    # results = Measurement.results.m_copy()
+    # results.section_def = XRDResult
```
Comment (hampusnasstrom) [inline src/nomad_measurements/xrd/schema.py:633]: Put this back

File: src/nomad_measurements/xrd/schema.py [hunk 4]
```diff
@@ -412,7 +661,7 @@ def normalize(self, archive: 'EntryArchive', logger: 'BoundLogger'):
                     incident_beam_wavelength=result.source_peak_wavelength,
                     two_theta_angles=result.two_theta,
                     intensity=result.intensity,
-                    q_vector=result.q_vector,
+                    # q_vector=result.q_vector,
```
Comment (hampusnasstrom) [inline src/nomad_measurements/xrd/schema.py:664]: Put this back

**Review by hampusnasstrom — CHANGES_REQUESTED** [review]
Good job! Just fix the q_norm thing.

File: src/nomad_measurements/xrd/IKZ.py [hunk 5]
```diff
@@ -186,12 +187,10 @@ def __init__(self, path, verbose=True):
...
-        self._ndscan = len(np.unique(list(map(len, data))))==1
-        if self._ndscan:
-            data = np.array(data)    
-        else:  # not originally part of Carsten's code 
-            raise NotImplementedError("Unequal lengths of multiple scans in a file not supported.")
-        imgdata = np.array(imgdata)
+        # self._ndscan = len(np.unique(list(map(len, data))))==1
+        # if self._ndscan:
+        #     data = np.array(data)
+        # imgdata = np.array(imgdata)
```
[the remark this answers is not in the document]
Comment (ka-sarthak) [reply src/nomad_measurements/xrd/IKZ.py:193]: Deleting the commented out code

File: src/nomad_measurements/xrd/schema.py [hunk 6]
```diff
@@ -268,31 +299,34 @@ def derive_n_values(self):
             return len(self.two_theta)
         return 0
 
+    scan_type = Quantity(
+        type=str,
+        description='Type of scan (line, multiline, rsm)',
+    )
```
[the remark this answers is not in the document]
Comment (ka-sarthak) [reply src/nomad_measurements/xrd/schema.py:305]: Yes, this is not required

**Review by hampusnasstrom — APPROVED** [review]
I tested it now and everything works. I have some suggestions on the plots but I will write a separate issue for this. For now I will go ahead and merge this.

**Pull request merged**